File: epi_judge_java/epi/EvenOddListMerge.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class EvenOddListMerge {

    @EpiTest(testDataFile = "even_odd_list_merge.tsv")
    public static ListNode<Integer> evenOddMerge(ListNode<Integer> head) {
        int i = 0;
        var runner = head;
        var leftDummy = new ListNode<>(-1, null);
        var leftRunner = leftDummy;
        var rightDummy = new ListNode<>(-1, null);
        var rightRunner = rightDummy;
        while (runner != null) {
            if (i % 2 == 0) {
                leftRunner.next = new ListNode<>(runner.data, null);
                leftRunner = leftRunner.next;
            } else {
                rightRunner.next = new ListNode<>(runner.data, null);
                rightRunner = rightRunner.next;
            }
            i++;
            runner = runner.next;
        }
        leftRunner.next = rightDummy.next;
        return leftDummy.next;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "EvenOddListMerge.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/even_odd_list_merge.cc
#include "list_node.h"
#include "test_framework/generic_test.h"

shared_ptr<ListNode<int>> EvenOddMerge(const shared_ptr<ListNode<int>> &list) {
    shared_ptr<ListNode<int>> evenDummy(new ListNode<int>);
    shared_ptr<ListNode<int>> oddDummy(new ListNode<int>);

    auto evenRunner = evenDummy;
    auto oddRunner = oddDummy;

    auto runner = list;

    int i = 0;
    while (runner != nullptr) {
        if (i % 2 == 0) {
            evenRunner->next = runner;
            evenRunner = evenRunner->next;
        } else {
            oddRunner->next = runner;
            oddRunner = oddRunner->next;
        }

        runner = runner->next;
        i++;
    }

    evenRunner->next = oddDummy->next;
    oddRunner->next = nullptr;

    return evenDummy->next;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"L"};
    return GenericTestMain(args, "even_odd_list_merge.cc",
                           "even_odd_list_merge.tsv", &EvenOddMerge,
                           DefaultComparator{}, param_names);
}
